server/src/immich/api-v1/asset/asset-repository.ts

```diff
@@ -360,4 +361,23 @@ export class AssetRepository implements IAssetRepository {
 
     return assetCountByUserId;
   }
+
+  /**
+   * Get asset by filepath on the database
+   * @param ownerId
+   * @param originalPath
+   *
+   */
+  getAssetByOriginalPath(ownerId: string, originalPath: string): Promise<AssetCheck[]> {
+    return this.assetRepository.find({
+      select: {
+        id: true,
+        checksum: true,
+      },
+      where: {
+        ownerId,
+        originalPath,
```

Comment:
If we are going to query by this i wonder if we need an index in this column. If we want it to be unique, we should consider a unique constraint.

Reply:
Added an index